File: contracts/rust/donatewithfeedback/test/consts.go
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package test

import (
	"github.com/iotaledger/wasp/packages/coretypes"
)

const ScName = "donatewithfeedback"
const ScHname = coretypes.Hname(0x696d7f66)

const ParamAmount = "amount"
const ParamFeedback = "feedback"

const VarAmount = "amount"
const VarDonations = "donations"
const VarDonator = "donator"
const VarError = "error"
const VarFeedback = "feedback"
const VarLog = "log"
const VarMaxDonation = "maxDonation"
const VarTimestamp = "timestamp"
const VarTotalDonation = "totalDonation"

const FuncDonate = "donate"
const FuncWithdraw = "withdraw"
const ViewDonations = "donations"

const HFuncDonate = coretypes.Hname(0xdc9b133a)
const HFuncWithdraw = coretypes.Hname(0x9dcc0f41)
const HViewDonations = coretypes.Hname(0x45686a15)

File: contracts/rust/donatewithfeedback/src/consts.rs
// Copyright 2020 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

#![allow(dead_code)]

use wasmlib::*;

pub const SC_NAME: &str = "donatewithfeedback";
pub const HSC_NAME: ScHname = ScHname(0x696d7f66);

pub const PARAM_AMOUNT: &str = "amount";
pub const PARAM_FEEDBACK: &str = "feedback";

pub const VAR_AMOUNT: &str = "amount";
pub const VAR_DONATIONS: &str = "donations";
pub const VAR_DONATOR: &str = "donator";
pub const VAR_ERROR: &str = "error";
pub const VAR_FEEDBACK: &str = "feedback";
pub const VAR_LOG: &str = "log";
pub const VAR_MAX_DONATION: &str = "maxDonation";
pub const VAR_TIMESTAMP: &str = "timestamp";
pub const VAR_TOTAL_DONATION: &str = "totalDonation";

pub const FUNC_DONATE: &str = "donate";
pub const FUNC_WITHDRAW: &str = "withdraw";
pub const VIEW_DONATIONS: &str = "donations";

pub const HFUNC_DONATE: ScHname = ScHname(0xdc9b133a);
pub const HFUNC_WITHDRAW: ScHname = ScHname(0x9dcc0f41);
pub const HVIEW_DONATIONS: ScHname = ScHname(0x45686a15);
